iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/ResourceVisitor.java

```diff
@@ -162,22 +163,33 @@ public Attribute reportIfFalse(String message, SecondaryLocation... secondaries)
       // designed to be extended but noop in standard case
       return this;
     }
-    public Attribute reportIfValueDoesNotMatch(String expectedValue, String message, SecondaryLocation... secondaries) {
+
+    public Attribute reportIfValueEquals(String expectedValue, String message, SecondaryLocation... secondaries) {
+      // designed to be extended but noop in standard case
+      return this;
+    }
+
+    public Attribute reportIfNotValueEquals(String expectedValue, String message, SecondaryLocation... secondaries) {
       // designed to be extended but noop in standard case
       return this;
     }
 
-    public Attribute reportIfValueMatches(String expectedValue, String message, SecondaryLocation... secondaries) {
+    public Attribute reportIf(Predicate<ExpressionTree> expectedPredicate, String message, SecondaryLocation... secondaries) {
       // designed to be extended but noop in standard case
       return this;
     }
 
-    public Attribute reportIfValueDoesNotMatch(Predicate<ExpressionTree> expectedPredicate, String message, SecondaryLocation... secondaries) {
+    public Attribute reportIfNot(Predicate<ExpressionTree> expectedPredicate, String message, SecondaryLocation... secondaries) {
       // designed to be extended but noop in standard case
       return this;
     }
```

Comment:
This method is only used in test cases. So it seems that there is no real use case so far.

Reply:
I've just renamed
`reportIfValueDoesNotMatch(Predicate<ExpressionTree> ...)` to
`reportIfNot(Predicate<ExpressionTree> ...)`
I did not plan to remove existing methods.
Or are do you mean some other method?

Reply:
Besides, if this one is missing, future users (i.e. we) will be encouraged to use
`reportIf(<predicate>.negate())`
which is both less readable and creates a new wrapper predicate on each invocation.
Due to this I suggest we keep it.